### DISCUSSION TITLE
ccache fails with xcode

### DISCUSSION THREAD

**Discussion by MMcKester** [ccache/ccache#498] [Support]
Hi, I added ccache to Xcode by the usual proposed way (overriding CC, etc). Ccache also gets invoked but in my simple **Hello World** example I never get a cache hit. Digging into the log files I discovered this error message:

```
[2019-12-10T18:45:58.510399 10589] Unlink /Volumes/.../path/to/file.o via /Volumes/...path/to/Objects-normal/x86_64/file.o.rm.my-hostname.10589.XXXXXX
[2019-12-10T18:45:58.510403 10589] x_unlink failed: No such file or directory
```

Does anyone know where this error message comes from or how to interpret it? Thank you!

**Comment by jrosdahl** [comment]
> I never get a cache hit.

See [_Cache debugging_](https://ccache.dev/manual/3.7.6.html#_cache_debugging) in the manual for hints on how to debug cache misses.

> x_unlink failed: No such file or directory
> [...]
> Does anyone know where this error message comes from or how to interpret it? Thank you!

The log message just says that ccache failed to delete a file, but you didn't include enough of the log for me to guess why. However, I think that it's unlikely that the message has something to do with not getting cache hits.
